src/dataqueue/queue.cc

```diff
@@ -162,6 +162,27 @@ class DataQueueImpl final : public DataQueue,
         "entries", entries_, "std::vector<std::unique_ptr<Entry>>");
   }
 
+  void addBackpressureListener(BackpressureListener* listener) override {
+    if (idempotent_) return;
+    DCHECK_NOT_NULL(listener);
+    backpressure_listeners_.insert(listener);
+  }
+
+  void removeBackpressureListener(BackpressureListener* listener) override {
+    if (idempotent_) return;
+    DCHECK_NOT_NULL(listener);
+    backpressure_listeners_.erase(listener);
+  }
+
+  void NotifyBackpressure(size_t amount) {
+    if (idempotent_) return;
+    for (auto listener : backpressure_listeners_) listener->EntryRead(amount);
+  }
+
+  bool has_backpressure_listeners() const {
```

Comment:
Nit: Why is this not a camel case?

Reply:
essentially treating it similar to a getter.